### ISSUE TITLE
Known issues with DB4S

### ISSUE THREAD

**Issue by chrisjlocke** [sqlitebrowser/sqlitebrowser#1688]
This issue is for tracking all known issues with DB Browser for SQLite (DB4S). It will show you what to expect when *using* and/or *building* DB4S.

It is meant to let you know that we are aware of these issues, but there are no fixes for them at the moment. Either because the fix is out of our control (e.g. It is a problem with a library we use) or we haven't figured out a way (yet!) to fix them. :smiley: 

Ideas, solutions, and PRs are always welcome.

-----
## macOS (only) - Error message: `The root CA certificate is not trusted for this purpose`

This is a known (minor) issue on macOS when trying to connect to DBHub.io via DB4S.

We're looking into it, and it's being tracked here: https://github.com/sqlitebrowser/sqlitebrowser/issues/2829

-----

**Comment by karim** [comment]
@chrisjlocke I'm just wondering why did you unpinn this issue? Is it okay to pin it back again?

**Comment by justinclift** [comment]
It was probably unpinned because we worked around the Qt issue by changing to Qt 5.11.3 for our nightly builds, and the OpenSSL 1.1 thing also only affects people compiling it themselves too.

The Wifi problem potentially affects more people, so just now pinned it back.  Good suggestion. :smile:

**Comment by karim** [comment]
I see, thanks.

We should keep this one open and pinned to list the things that are out of our control and things we cannot really fix, but maybe provide a workaround till they are fixed by the affected library.

**Comment by justinclift** [comment]
Just changed removed the WiFi entry, as the fix is now in our most recent release (3.11.1).  Also updated the title to mention "when compiling" as both of the remaining issues are developer only items (at this point).  Naturally, this can be changed around later depending if/what/how/when/where (etc) new significant issues show up. :smile:

**Comment by karim** [comment]
Good point on removing the WiFi part since it is fixed now. :+1: 

I think we should split the first post into two sections; when *using* and the other when *compiling* DB4S.

I see that the dark mode/theme does not work as intended. Many issues are opened for that. On macOS and Windows 10, right?

Should this be noted here as well since this is a Qt thing we cannot do anything about, and so that we can close all new issues about it with a link to this issue?

**Comment by justinclift** [comment]
Oh yeah, good approach.  I'm about to get off the computer, so feel free to get that done if you want.

Otherwise, I'll get to it tomorrow. :smile:

**Comment by GortiZ6** [comment]
Long time no see... sorry! Anyway I've tried to build the latest git with Qt 5.12.3..
![immagine](https://user-images.githubusercontent.com/3738577/58862194-69dd7d00-86b0-11e9-8042-c57a3c4f3eda.png)
Is this really solved or it's again just me testing it in the wrong way?

**Comment by justinclift** [comment]
Oh cool, looks like it might be solved now after all! :smile:

**Comment by chrisjlocke** [comment]
Shouldn't it be on one line, rather than wordwrapped on 3 lines?  Guess
thats an application option...  does that work @GortiZ?

>

**Comment by GortiZ6** [comment]
I enlarged the line to test. This is with the standard 1 line height:
![image](https://user-images.githubusercontent.com/3738577/58866103-25ee7600-86b8-11e9-8a4d-ddc15c59b84f.png)

**Comment by chrisjlocke** [comment]
Aah, cool.  That  looks more 'normal' to me.  Excellent. \o/

**Comment by justinclift** [comment]
I'll update the Qt version on our nightly build server VMs over the next few days. :smile:

**Comment by mgrojo** [comment]
It should also do automatic word wrapping when the line is very long and has embedded spaces.

**Comment by justinclift** [comment]
Looks like your bug report was worth the effort @mgrojo. :grin:

**Comment by GortiZ6** [comment]
@mgrojo yes, I can confirm also a single line is properly word wrapped. This was windows and I've not checked yet for Linux, but Qt should behave the same on every platform, so your bug report seems was accepted and fixed ;)

**Comment by MKleusberg** [comment]
What do you think? Can we close this issue? Both issues regarding dark mode are closed and I believe the macOS problem should be gone since we notarize our builds.

**Comment by chrisjlocke** [comment]
Yes, I think it was just a 'general place' for any issues that were show-stopppers...

**Comment by justinclift** [comment]
Cool, closing this then and unpinning it. :smile:

**Comment by lucydodo** [comment]
@chrisjlocke Is there a reason pinned it again? 🤔

**Comment by chrisjlocke** [comment]
Yes, it was unpinned, but Justin wanted to add to it, and couldn't find it, so I pinned it again, and then he found it, then he edited it.  The last part in it is current.

**Comment by lucydodo** [comment]
@chrisjlocke oh, I got it. Thanks for response. 😄

**Comment by lucydodo** [comment]
and one more, how about you think reopen this issue?

**Comment by justinclift** [comment]
@lucydodo Good idea. :smile:

**Comment by lucydodo** [comment]
Hi @chrisjlocke, Do you mind if I edit the content of this issue a bit?

Our CI workflow sometimes fails. If the CI workflow fails for a particular commit or PR, I can leave a comment on the commit or PR to track and resolve the issue, but the downside is that if it works on a schedule, like a nightly build, there is no place to leave a comment and I have to create a new issue or discussion.

Of course, I can email @justinclift privately to track and resolve the issue when disconnected from our build server,
but otherwise I want to track it together here. 😄

**Comment by justinclift** [comment]
@lucydodo You don't need to ask, you're welcome to do it. :smile:

**Comment by lucydodo** [comment]
Thank you, I was worried it might be rude to edit it arbitrarily. :)
